src/api/secrets.ts

```diff
@@ -1,30 +1,28 @@
 import { createRadixApiUrl } from './api-config';
-import { getJson, putJson } from './api-helpers';
+import { putJson } from './api-helpers';
 
-export async function getComponentSecret(namespace, componentName) {
-  return await getJson(
-    createRadixApiUrl(`namespaces/${namespace}/${componentName}`)
-  );
-}
+import { SecretType } from '../models/secret-type';
 
 export async function saveComponentSecret(
-  appName,
-  envName,
-  componentName,
-  secretName,
-  value
-) {
+  appName: string,
+  envName: string,
+  componentName: string,
+  secretName: string,
+  value: string
+): Promise<string> {
   const encAppName = encodeURIComponent(appName);
   const encEnvName = encodeURIComponent(envName);
   const encComponentName = encodeURIComponent(componentName);
   const encSecretName = encodeURIComponent(secretName);
 
-  const body = { secretValue: value.toString() };
+  const body: { secretValue: string; type?: SecretType } = {
```

Comment:
Why is type defined when it is not used?

Reply:
It's a valid field for this request for the API, so I just added it

Reply:
I checked radix-api, and the field is declared, but never used. It was added by Sergey 11 months ago. We'll ask him when he returns from vacation. I think we can safely remove the type field from secrets.ts